### ISSUE TITLE
Potentially wrong Bounding Box augmentation using imgaug

### ISSUE THREAD

**Issue by pradeeprathore04** [matterport/Mask_RCNN#968]
It seems that the bounding box augmentation will not be correct in case of rotation using imgaug. Since we have to supply train_generator and val_generator in model.train. I am using the below code for rotation of images as well as bounding box
augmentation = iaa.Sequential([ iaa.OneOf([ iaa.Affine(rotate=(-5, 5)])
model.train(dataset_train, dataset_val, 
            learning_rate=config.LEARNING_RATE, 
            epochs=NUM_EPOCHS, 
            layers='all',
            augmentation=augmentation.to_deterministic()))

Whereas using imgaug, the bounding box and image augmentation should be done as below
image_aug = seq_det.augment_images([image])
bbs_aug = seq_det.augment_bounding_boxes([bbs])

If I will supply one augmentation, I am not sure whether the current implementation will doing the bounding box augmentation correctly since it is internally using Keras data_generator in model.py

        # Data generators
        train_generator = data_generator(train_dataset, self.config, shuffle=True,
                                         augmentation=augmentation,
                                         batch_size=self.config.BATCH_SIZE,
                                         no_augmentation_sources=no_augmentation_sources)
        val_generator = data_generator(val_dataset, self.config, shuffle=True,
                                       batch_size=self.config.BATCH_SIZE)

If anybody can suggest how can I use bounding box and image augmentation correctly here, I would be grateful. Thanks a lot.
